muxtools/subtitle/font.py

```diff
@@ -143,8 +192,115 @@ def clean_name(fontname: str) -> str:
 
             if not outpath.exists():
                 shutil.copy(fontpath, outpath)
+            
+            if subset_fonts:
+                # We always want to include the used characters, and any additional glyphs specified by the user
+                characters_set = usage_data.characters_used
+                characters_set.update(subset_additional_glyphs_parsed)
+
+                if not subset_aggressive:
+                    # We also want to add a general subset of common characters to avoid issues when reusing fonts, if we aren't doing aggressive subsetting
+                    characters_set.update(COMMON_UNICODE_CHARS)
+
+                if not subset_ignore_fonts_with_no_usage and not characters_set:
+                    # If theres no characters used, and we aren't ignoring fonts with no usage, add common characters
+                    warn(f"No characters used in font '{fontname}'. Defaulting to common subset.", collect_fonts)
+                    characters_set.update(COMMON_UNICODE_CHARS)
+
+                if not characters_set:
+                    warn(f"No characters used in font '{fontname}'. Skipping subsetting...", collect_fonts)
+                
+                else:
+                    new_path = outpath.with_name(f"{outpath.stem}_subset{outpath.suffix}")
+
+                    old_font_family_name = ''
+                    new_font_family_name = ''
+                    old_font_name = ''
+                    new_font_name = ''
+
+                    try:
+                        font = ttLib.TTFont(outpath)
+
+                        subsetter = Subsetter()
+                        subsetter.populate(text="".join(characters_set))
+                        subsetter.subset(font)
+
+                        name_table = font["name"]
+
+                        for record in name_table.names:
+                            # Font names can only be 31 characters long
+
+                            if record.nameID == 1:  # Font Family name
+                                old_font_family_name = record.toUnicode().strip()
+                                new_font_family_name = record.toUnicode().replace(' ', '')[:25] + "Subset"
+                                record.string = new_font_family_name
+                            elif record.nameID == 4:  # Full name
+                                old_font_name = record.toUnicode().strip()
+                                new_font_name = record.toUnicode().replace(' ', '')[:25] + "Subset"
+                                record.string = new_font_name
+                            elif record.nameID == 6:  # PostScript name
+                                record.string = record.toUnicode().replace(' ', '')[:25] + "Subset"
```

Comment:
it *might* be better to always set the postscript name to equal the full name instead? (this will usually, but not always, be the case)

Reply:
I don’t see why you’d want to do that tbh, and in my experience they are _usually_ different. Full names usually contain spaces; PostScript names are _forbidden_ from containing spaces (although neither GDI nor libass care).

Reply:
actually, good point. i think i was mixing up something else in my mind?
anyway, the later point about “what if \fn uses a postscript name” stands, as that's not registered for replacement anywhere